textpipe/doc.py

```diff
@@ -249,6 +250,18 @@ def words(self):
 
         return [(token.text, token.idx) for token in self._spacy_doc]
 
+    @property
+    def word_counts(self):
+        """
+        Extract words with their counts
+
+        >>> doc = Doc('Test sentence for testing vectorisation of a sentence.')
+        >>> doc.word_counts
+        {'Test': 1, 'sentence': 2, 'for': 1, 'testing': 1, 'vectorisation': 1, 'of': 1, 'a': 1, '.': 1}
+        """
+
+        return dict(Counter(word for word, _ in self.words))
```

Comment:
This:
```
def s4(mylist):
    mydict={}.fromkeys(mylist,0)
    for k in mydict:
        mydict[k]=mylist.count(k)    
    return mydict
```

or this:
```
def s6(mylist):
    mydict=defaultdict(int)
    for i in mylist:
        mydict[i] += 1
    return mydict     
```
Is supposed to be substantially faster. 

Worth considering, maybe?

Source: https://stackoverflow.com/questions/18343472/efficient-way-to-count-the-element-in-a-dictionary-in-python-using-a-loop

Reply:
it's not orders of magnitude faster: I'm in favor of clean code.

Reply:
Fair point, OK